Packet++/header/S7commLayer.h

```diff
@@ -0,0 +1,234 @@
+		 * @param[in] error_class The value of the error class
+		 * @param[in] error_code The value of the error code
+		 */
+		S7commLayer(uint8_t msg_type, uint16_t pdu_ref, uint16_t param_length, uint16_t data_length,
+					uint8_t error_class = 0, uint8_t error_code = 0);
+
+		/**
+		 * A constructor that creates the layer from an existing packet raw data
+		 * @param[in] data A pointer to the raw data (will be casted to @ref s7commhdr)
+		 * @param[in] dataLen Size of the data in bytes
+		 * @param[in] prevLayer A pointer to the previous layer
+		 * @param[in] packet A pointer to the Packet instance where layer will be stored in
+		 */
+		S7commLayer(uint8_t *data, size_t dataLen, Layer *prevLayer, Packet *packet)
+			: Layer(data, dataLen, prevLayer, packet)
+		{
+			m_Protocol = S7COMM;
+			m_Parameter = nullptr;
+		}
+
+		virtual ~S7commLayer() { delete m_Parameter; }
+
+		/**
+		 * @return S7comm protocol id
+		 */
+		uint8_t getProtocolId() const;
+
+		/**
+		 * @return S7comm message type
+		 */
+		uint8_t getMsgType() const;
+
+		/**
+		 * @return S7comm reserved
+		 */
+		uint16_t getReserved() const;
+
+		/**
+		 * @return S7comm PDU ref
+		 */
+		uint16_t getPduRef() const;
+
+		/**
+		 * @return S7comm parameter length
+		 */
+		uint16_t getParamLength() const;
+
+		/**
+		 * @return S7comm data length
+		 */
+		uint16_t getDataLength() const;
+
+		/**
+		 * @return S7comm error code
+		 */
+		uint8_t getErrorCode() const;
+		/**
+		 * @return S7comm error class
+		 */
+		uint8_t getErrorClass() const;
+
+		const S7CommParameter *getParameter();
+
+		/**
+		 * Set the value of the message type
+		 * @param[in] msg_type The value of the message type
+		 */
+		void setMsgType(uint8_t msg_type) const;
+
+		/**
+		 * Set the value of the PDU ref
+		 * @param[in] pdu_ref The value of the PDU ref
+		 */
+		void setPduRef(uint16_t pdu_ref) const;
+
+		/**
+		 * Set the value of the parameter length
+		 * @param[in] param_length The value of the parameter length
+		 */
+		void setParamLength(uint16_t param_length) const;
+
+		/**
+		 * Set the value of the data length
+		 * @param[in] data_length The value of the data length
+		 */
+		void setDataLength(uint16_t data_length) const;
+
+		/**
+		 * Set the value of the error code
+		 * @param[in] error_code The value of the error code
+		 */
+		void setErrorCode(uint8_t error_code) const;
+		/**
+		 * Set the value of the error class
+		 * @param[in] error_class The value of the error class
+		 */
+		void setErrorClass(uint8_t error_class) const;
+
+		/**
+		 * @return Size of @ref s7commhdr
+		 */
+		size_t getHeaderLen() const override
+		{
+			if (getS7commHeader()->msg_type == 0x03)
+				return sizeof(s7comm_ack_data_hdr) + getParamLength() + getDataLength();
+
+			return sizeof(s7commhdr) + getParamLength() + getDataLength();
+		}
+
+		/**
+		 * Does nothing for this layer
+		 */
+		void computeCalculateFields() override {}
+
+		/**
+		 * Does nothing for this layer
```

Comment:
Let's mention that this is always the last layer?
`Does nothing for this layer (S7CommLayer is always last)`